marketplace-service/src/main/java/com/axonivy/market/github/service/impl/GHAxonIvyProductRepoServiceImpl.java

```diff
@@ -213,10 +221,31 @@ private void getDependencyContentsFromProductJson(ProductModuleContent productMo
         productModuleContent.setArtifactId(artifact.getArtifactId());
         productModuleContent.setType(artifact.getType());
         productModuleContent.setName(artifact.getName());
+
+
+        boolean isProductJsonContentNotFound = Optional.ofNullable(
+            productJsonContentRepository.findByNameAndTag(product.getId(), productModuleContent.getTag())).isEmpty();
```

Comment:
since it is checking an existing boolean, to save performance, let's create a query to check if the target record exists or not from the db layer rather than requiring mongo to return the whole object and check that result.